prov/sm2/src/sm2_progress.c

```diff
@@ -105,10 +106,14 @@ static int sm2_start_common(struct sm2_ep *ep,
 			total_len, comp_buf, xfer_entry->hdr.sender_gid,
 			xfer_entry->hdr.tag, xfer_entry->hdr.cq_data);
 	}
+
 	if (ret) {
+		xfer_entry->hdr.op_flags &= ~FI_DELIVERY_COMPLETE;
```

Comment:
I don't think I understand this one. If the sender requested delivery complete, then don't you need that to trigger the tx completion, even on error? How do you send back the completion status? Ie in this case, the receive failed so the sender should have an error entry written with a reason of why it failed.

Reply:
We don't communicate it back right now

Reply:
I think that's the bigger bug here. Adding this would just be replacing one bug (communicating success on the send when it was actually an error) with another (never writing a send completion on error). I would even prefer for the send to complete in success in this case since the message did arrive to the receiver.